myia/operations/macro_einsum.py

```diff
@@ -0,0 +1,181 @@
+"""Implementation of the 'einsum' macro."""
+
+from opt_einsum import contract, contract_expression
+
+from ..lib import InferenceError, macro
+from . import primitives as P
+
+
+def _tensordot(g, a, b, *, axes):
```

Comment:
Could you write some light documentation for this and the other private functions in this module? It'd help a little for review/maintenance.

Reply:
Done